packages/nimble-components/src/rich-text/viewer/tests/rich-text-viewer.stories.ts

```diff
@@ -32,6 +34,13 @@ const metadata: Meta<RichTextViewerArgs> = {
     <${richTextViewerTag}
         :markdown="${x => x.markdown}"
     >
+        <${richTextMentionUsersTag} pattern="^user:(.*)">
```

Comment:
While we currently use this pattern exclusively for user mentions, it holds potential for future scenarios. For instance, as Milan pointed out in [this comment](https://github.com/ni/nimble/pull/1594#discussion_r1364736273), links directing to an issue are presently absolute links but will be converted into issue mention view elements once support is added. 

Also, the user mentions lack of a specific use case for having a pattern, but looking ahead, if a client wishes to implement a user link that, upon clicking, navigates to the user's profile, we may require a pattern allowing the client to determine the link in configuration element.

Reply:
> links directing to an issue are presently absolute links but will be converted into issue mention view elements once support is added.

Sure, but we would not re-use the `nimble-rich-text-mention-users` element for issues, so I'm still unsure why we would need to configure the pattern on the user mentions element.

> we may require a pattern allowing the client to determine the link in configuration element.

This sounds like a new feature that we would have to add support for in Nimble, and we could update the pattern used to something like `^user:([^:]*)(::.*)?` to support appending an optional url. I'm just thinking about this from a client's perspective, and I'm confused about why the pattern is configurable instead of just documented. Would I ever want to use a different pattern in different viewer/editor instances? Can I just use the pattern `'user'` or even `''`? What are the implications of my pattern choice?

At the least, it would probably be good to export a constant for `'^user:(.*)` and recommend using that.

@rajsite I'm curious to hear your thoughts on this.

Reply:
The idea was not to be coupled to a specific system's representation of user hrefs.
SystemLink doesn't have user profile pages but if it did, at like `https://test.lifecyclesolutions.ni.com/users/milan` then you would want a regex that can match that url pattern, something like `http:\/\/.*/users\/.*`
On the other side of the fence they are choosing to use `user:` I think, but that's not a requirement of all `rich-text-mention-users` users.
We are also not precluded from making a SystemLink specific `rich-text-systemlink-mention-users` type thing in the future that keeps the pattern hard coded and internal.

Reply:
Could we add some test coverage for the case where the mention regex matches certain links that start with http:// or https://? It'd be good to include this in both storybook and unit tests. In storybook it could be a selector to set different match patterns.

Reply:
Added tests for https:// and http:// links as mention node and also as a link in markdown-parser.spec.ts file.
Included in the storybook docs as a selector between `https://user/.*` and `user:.*`